docs/signing-spec.md

```diff
@@ -129,17 +158,69 @@ label.size = 248098
 
 This format does not change with groups or conditions.
 
-The signature is computed by concatenating the following pieces of data together in a line-separated (`\n`) UTF-8 string: `by`, `name`, `version`, and the `label.sha256` of each parcel:
+The signature is computed by concatenating the following pieces of data together in a line-separated (`\n`) UTF-8 string: `by`, `name`, `version`, `role`, and the `label.sha256` of each parcel:
 
 ```
 Matt Butcher <matt.butcher@example.com>
 mybindle
 0.1.0
+creator
+~
 e1706ab0a39ac88094b6d54a3f5cdba41fe5a901
 098fa798779ac88094b6d54a3f5cdba41fe5a901
 5b992e90b71d5fadab3cd3777230ef370df75f5b
 ```
 
+Note that the sequence `\n~\n` is used as a separator to prevent an attempt to conflate a 
+SemVer with the SHA list.
+
+> TODO: Do we need/want a timestamp on the signature?
```

Comment:
I wonder if we add an optional signing metadata section to support this. It could be really useful information for some people, but I don't think it warrants inclusion within the signing block itself

Reply:
I agree with an optional metadata section, but why would the timestamp be optional?

(Including it could allow for things like purging bindles with signatures older than a particular timestamp.)

Reply:
If a bindle _has_ a metadata section, it MUST have a timestamp. Otherwise, it doesn't need one

Reply:
Alright, so after talking with both of you, I have made Timestamp a _required_ field. I have also made it a u64 to avoid the kind of nonsense we have with Kubernetes and Go's not-quite-RFC-compliant date formatters. A timestamp is easy to parse, as close to unambiguous as anything in our industry gets, and supported across just about everything.